docs: add new audit log docs (#1847)
rancher/rancher-docs: docs/troubleshooting/other-troubleshooting-tips/user-id-tracking-in-audit-logs.md

## Feature Description

- When Kubernetes Audit logs are enabled on the downstream cluster, in each event that is logged, the external Identity Provider's username is now logged for each request, at the "metadata" level.

> - When Rancher API Audit logs are enabled on the Rancher installation, the external Identity Provider's username is also logged now at the `auditLog.level=0` for each request that hits the Rancher API server, including the login requests.

Reviewer: ```suggestion
- When you enable Rancher API Audit logs for a Rancher installation, the external Identity Provider's username is also logged now at the `auditLog.level=0` for each request that hits the Rancher API server, including login requests.
```
